sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/CosmosAuthTypes.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.kafka.connect.implementation;
+
+public enum CosmosAuthTypes {
+    MASTER_KEY("MasterKey"),
+    SERVICE_PRINCIPAL("ServicePrincipal");
+
+    private final String name;
+
+    CosmosAuthTypes(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public static CosmosAuthTypes fromName(String name) {
+        for (CosmosAuthTypes authTypes : CosmosAuthTypes.values()) {
+            if (authTypes.getName().equalsIgnoreCase(name)) {
+                return authTypes;
+            }
+        }
+        return null;
```

Comment:
why return null and not error out? This is all implementation details, and we should be the only ones changing, right?

Reply:
Actually the validation of authConfig should take care of this, thoughts?

Reply:
yea `AuthTypeValidator` will validate it - this happens during the config def validation